observability-test/observability.test.ts

```diff
@@ -0,0 +1,417 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const assert = require('assert');
+const {
+  AlwaysOffSampler,
+  AlwaysOnSampler,
+  NodeTracerProvider,
+  InMemorySpanExporter,
+} = require('@opentelemetry/sdk-trace-node');
+const {SpanStatusCode} = require('@opentelemetry/api');
+const {SimpleSpanProcessor} = require('@opentelemetry/sdk-trace-base');
+const {
+  SPAN_NAMESPACE_PREFIX,
+  getActiveOrNoopSpan,
+  setSpanError,
+  setSpanErrorAndException,
+  startTrace,
+} = require('../src/instrument');
+const {
+  SEMATTRS_DB_NAME,
+  SEMATTRS_DB_SQL_TABLE,
+  SEMATTRS_DB_STATEMENT,
+  SEMATTRS_DB_SYSTEM,
+  SEMATTRS_EXCEPTION_MESSAGE,
+} = require('@opentelemetry/semantic-conventions');
+
+const {disableContextAndManager, setGlobalContextManager} = require('./helper');
+
+const {
+  AsyncHooksContextManager,
+} = require('@opentelemetry/context-async-hooks');
+
+const projectId = process.env.SPANNER_TEST_PROJECTID || 'test-project';
+
+describe('Test startTrace', () => {
```

Comment:
Hey @surbhigarg92, I purposefully setup this directory because running all the tests locally is a huge chore, given that some of them require mutating a live Cloud Spanner instance, and they are unrelated to observability, which makes development so much harder. It also helps to have an accurate view of coverage.

Reply:
Also as you can see, testing for observability is a very involved process for which it is so much easier when its tests are isolated and can be focused on to test rigorously than when mixed with the other functional tests that are related to Cloud Spanner correctness/functionality.